src/test/java/gui/controllers/RoomControllerTest.java

```diff
@@ -0,0 +1,304 @@
+        controller.itemsRemoved(itemsDone);
+
+        assertEquals(2, controller.getProgress().getRoom().getChestsOpened());
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test updateActivity method.
+     */
+    @Test
+    void updateActivityTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Label label = new Label();
+        controller.setActivityStatus(label);
+
+        controller.updateActivity();
+        assertTrue(label.getText().contains("zero"));
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test updateChests method.
+     */
+    @Test
+    void updateChestsTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Label label = new Label();
+        controller.setNumOfChestsOpened(label);
+
+        controller.updateChests(1);
+        assertTrue(label.getText().contains("1/3"));
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test changeInformation method.
+     */
+    @Test
+    void changeInformationTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Label label = new Label();
+        controller.setGameStatus(label);
+        camHandler.setBeginTime(0);
+        controller.setProgressBar(new GridPane());
+
+        controller.changeInformation(1);
+        assertTrue(label.getText().contains("started"));
+
+        controller.getProgress().getRoom().setTargetDuration(0);
+        controller.changeInformation(1);
+        assertTrue(label.getText().contains("ended"));
+
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test updateWarning method.
+     */
+    @Test
+    void updateWarningPaneTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Pane pane = new Pane();
+        Pane pane2 = new Pane();
+        pane.getChildren().add(new Pane());
+        pane.getChildren().add(new Pane());
+        pane.getChildren().add(pane2);
+        controller.setStatusPane(pane);
+
+        controller.updateWarningPane();
+        assertFalse(pane2.isVisible());
```

Comment:
Waarom assert je alleen maar pane2?

Reply:
omdat die visible moet zijn nadat update warning pane is aangeroepen.